src/EditorFeatures/Core/Implementation/IntelliSense/SignatureHelp/Controller.Session_ComputeModel.cs

```diff
@@ -158,17 +172,18 @@ private static SignatureHelpItem GetSelectedItem(Model currentModel, SignatureHe
             }
 
             private static bool DisplayPartsMatch(SignatureHelpItem i1, SignatureHelpItem i2)
-            {
-                return i1.GetAllParts().SequenceEqual(i2.GetAllParts(), CompareParts);
-            }
+                => i1.GetAllParts().SequenceEqual(i2.GetAllParts(), CompareParts);
 
             private static bool CompareParts(TaggedText p1, TaggedText p2)
-            {
-                return p1.ToString() == p2.ToString();
-            }
-
-            private async Task<Tuple<ISignatureHelpProvider, SignatureHelpItems>> ComputeItemsAsync(
-                IList<ISignatureHelpProvider> providers,
+                => p1.ToString() == p2.ToString();
+
+            /// <summary>
+            /// Returns <code>null</code> if our work was preempted and we want to return the 
+            /// previous model we've computed.
+            /// </summary>
+            private async Task<(ISignatureHelpProvider provider, SignatureHelpItems items)?> ComputeItemsAsync(
+                int id,
```

Comment:
Can we call this retriggerId or something so it's clear it relates to the other field? `id` is a bit vague.